applications/FemToDemApplication/custom_processes/generate_dem_process.cpp

```diff
@@ -132,14 +132,24 @@ void GenerateDemProcess::CreateDEMParticle(
     rNode.SetValue(IS_DEM, true);
     rNode.SetValue(RADIUS, Radius);
     rNode.SetValue(DEM_PARTICLE_POINTER, spheric_particle);
+
+    // We transfer kinematic information
+    if (rNode.SolutionStepsDataHas(VELOCITY)) {
```

Comment:
This is way inefficient, if your algorithm allows so, I'd check it once for the first node in the mesh.

Reply:
that's definetly true